include/revng/Model/Register.h

```diff
@@ -327,9 +514,54 @@ inline size_t getSize(Values V) {
   }
 }
 
-} // namespace model::Register
+template<model::Architecture::Values Architecture>
+constexpr model::Register::Values getFirst() {
+  if constexpr (Architecture == model::Architecture::x86)
+    return model::Register::eax_x86;
+  else if constexpr (Architecture == model::Architecture::arm)
+    return model::Register::r0_arm;
+  else if constexpr (Architecture == model::Architecture::mips)
+    return model::Register::v0_mips;
+  else if constexpr (Architecture == model::Architecture::mipsel)
+    return model::Register::v0_mips;
+  else if constexpr (Architecture == model::Architecture::x86_64)
+    return model::Register::rax_x86_64;
+  else if constexpr (Architecture == model::Architecture::aarch64)
+    return model::Register::x0_aarch64;
+  else if constexpr (Architecture == model::Architecture::systemz)
+    return model::Register::r0_systemz;
+  else
+    static_assert(value_always_false<Architecture>::value,
+                  "Unsupported architecture");
+}
```

Comment:
It's easy to introduce an error in this function.

Usually you have something like:

```c++
enum {
  Group1_Start,
  Group1_A = Group1_Start,
  Group1_B,
  Group1_C,
  Group1_End = Group1_C,
  Group2_Start,
  Group2_A = Group2_Start,
  Group2_B,
  Group2_C,
  Group2_End = Group2_C
};
```

But the autogenerated stuff prevents us from doing these things. We'll have to expand it to support grouping of enum items.